packages/runtime/container-runtime/src/test/runningSummarizer.spec.ts

```diff
@@ -647,6 +670,69 @@ describe("Runtime", () => {
                     ]), "unexpected log sequence 3");
                 });
             });
+
+            describe("Disabled Heuristics", () => {
+                it("Should not summarize after time or ops", async () => {
+                    await startRunningSummarizer({ disableHeuristics: true });
+
+                    await emitNextOp(summaryConfig.maxOps + 1);
+                    assertRunCounts(0, 0, 0, "should not summarize after maxOps");
+
+                    await tickAndFlushPromises(summaryConfig.idleTime + 1);
+                    assertRunCounts(0, 0, 0, "should not summarize after idleTime");
+
+                    await tickAndFlushPromises(summaryConfig.maxTime + 1);
+                    assertRunCounts(0, 0, 0, "should not summarize after maxTime");
+
+                    await emitNextOp(summaryConfig.maxOps * 3 + 10000);
+                    await tickAndFlushPromises(summaryConfig.maxTime * 3 + summaryConfig.idleTime * 3 + 10000);
+                    assertRunCounts(0, 0, 0, "make extra sure");
+                });
+
+                it("Should not summarize before closing", async () => {
+                    await startRunningSummarizer({ disableHeuristics: true });
+
+                    await emitNextOp(51); // hard-coded to 50 for now
+                    const stopP = summarizer.waitStop();
+                    await flushPromises();
+                    await emitAck();
+                    await stopP;
+
+                    assertRunCounts(0, 0, 0, "should not perform lastSummary");
+                });
+
+                it("Should not summarize immediately if summary ack is missing at startup when disabled", async () => {
+                    assertRunCounts(0, 0, 0);
+                    // Simulate as summary op was in op stream.
+                    const summaryTimestamp = Date.now();
+                    emitBroadcast(summaryTimestamp);
+
+                    let startStatus: "starting" | "started" | "failed" = "starting";
+                    startRunningSummarizer({ disableHeuristics: true }).then(
```

Comment:
I am not sure when the status goes from "starting" to "started". If we create a new RunningSummarizer, shouldn't the status always be "started"? Or does it move to started when it actually attempts to generate a summary?

Reply:
So I could probably have left this check out of this test, since it's already covered under the above test: "Should summarize immediately if summary ack is missing at startup". I left it since these are more like functional tests rather than unit tests, and it was just a harmless copy-paste.

As for how this check works: RunningSummarizer has a concept of `waitStart()` that resolves once the RunningSummarizer starts running its heuristics. So `RunningSummarizer.start()` will actually create a new RunningSummarizer instance, and then call `waitStart()` and only resolve when it completes.

It is waiting for the underlying SummaryCollection to flush its pending summaries. So basically don't start yet until we see any outstanding summary ops receive an ack/nack (or timeout). The test covers this waiting by the following:

1. Initialize `startStatus` to "starting".
2. Call RunningSummarizer.start(), which calls waitStart().
3. Attach Promise.then such that as soon as waitStart() completes:
    - we set `startStatus` to "started" when the promise resolves
    - we set `startStatus` to "failed" if the promise rejects
4. Wait for all promises to flush, so that the microtask queue clears out (potentially calling the Promise.then above).
5. Verify that `startStatus` still remains unchanged: "starting", because we have an outstanding summary op.
6. Emit an ack for that summary op and wait for promises to flush again.
7. Verify that `startStatus` is "started"; now we know waitStart() resolved at the expected point in time.

Reply:
Ah. So it's basically testing that the RunningSummarizer doesn't start if there are pending summary ops. I thought it is waiting for a summary to be attempted and hence the confusion.
Maybe renaming the test to say "Should not start summarizer immediately if .." will help.